qiskit/quantum_info/operators/symplectic/clifford_circuits.py

```diff
@@ -174,6 +202,131 @@ def _append_operation(clifford, operation, qargs=None):
     raise QiskitError(f"Cannot apply {gate}")
 
 
+def _prepend_operation(clifford, operation, qargs=None):
+    """Update Clifford inplace by prepending a Clifford operation.
+
+    Args:
+        clifford (Clifford): The Clifford to update.
+        operation (Instruction or Clifford or str): The operation or composite operation to apply
+           before the clifford.
+        qargs (list or None): The qubits to apply operation to.
+
+    Returns:
+        Clifford: the updated Clifford.
+
+    Raises:
+        QiskitError: if input operation cannot be converted into Clifford operations.
+    """
+    # pylint: disable=too-many-return-statements
+    if isinstance(operation, (Barrier, Delay)):
+        return clifford
+
+    if qargs is None:
+        qargs = list(range(clifford.num_qubits))
+
+    gate = operation
+
+    if isinstance(gate, str):
+        # Check if gate is a valid Clifford basis gate string
+        if gate not in _BASIS_PREP_1Q and gate not in _BASIS_PREP_2Q:
+            raise QiskitError(f"Invalid Clifford gate name string {gate}")
+        name = gate
+    else:
+        # assert isinstance(gate, Instruction)
+        name = gate.name
+        if getattr(gate, "_condition", None) is not None:
+            raise QiskitError("Conditional gate is not a valid Clifford operation.")
```

Comment:
Do we need this branch - didn't we remove the support for `_condition` a long time ago? (I understand that this function is adapted from `_append_iteration`, so maybe we should remove a similar code block there as well?)